make/github.mk

```
@@ -0,0 +1,19 @@
+# github.mk
+# Helpers for fetching released binaries from github projects.
+
+# For usage instructions, see uses of this macro elsewhere, since there is no
+# single standard for architecture naming.
+GET_GITHUB_RELEASE_FN = get_github_release() { \
+	[ -x $${1} ] || { \
+		set -euo pipefail ;\
+		echo "+ $${1}" ;\
+		mkdir -p bin ;\
+		attempts=5 ;\
+		for i in $$(seq $$attempts); do \
+			curl --silent --show-error --fail --location --output $${1} $${2} && break ;\
+			[ $$i -eq $$attempts ] && exit 1; \
```

Thinking aloud: let's maybe add 1-5 seconds sleep between attempts?
```suggestion
			curl --silent --show-error --fail --location --output $${1} $${2} && break ;\
			sleep 2 ;\
			[ $$i -eq $$attempts ] && exit 1; \
```

---

Good point. I added exponential backoff :wink: